File: example/java/spring-boot/two_consumers_one_provider/common-person-consumer/src/main/java/br/com/zup/pact/commonperson/dto/ClientDetailsDTO.java
package br.com.zup.pact.commonperson.dto;

import br.com.zup.pact.commonperson.entity.Client;
import java.util.Objects;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ClientDetailsDTO {

    private Integer id;
    private Integer accountId;
    private String name;
    private String finalName;
    private Integer age;

    public static Client fromDtoToEntity(ClientDetailsDTO clientDetailsDTO) {
        if (Objects.nonNull(clientDetailsDTO)) {
            return Client.builder()
                    .id(clientDetailsDTO.getId())
                    .accountId(clientDetailsDTO.getAccountId())
                    .name(clientDetailsDTO.getName())
                    .finalName(clientDetailsDTO.getFinalName())
                    .age(clientDetailsDTO.getAge())
                    .build();

        }
        return null;
    }
}

File: example/kotlin/one-consumer-one-provider/consumer/consumer/src/main/kotlin/br/com/zup/pact/consumer/dto/ClientDetailsDTO.kt
package br.com.zup.pact.consumer.dto

import br.com.zup.pact.consumer.entity.Client
import java.util.*

class ClientDetailsDTO (
        private val id: Int,
        private val accountId: Int,
        private val name: String,
        private val finalName: String,
        private val age: Int ) {

    fun fromDtoToEntity(clientDetailsDTO: ClientDetailsDTO): Client? {
        if (Objects.nonNull(clientDetailsDTO)) {
            return Client(
                    id = clientDetailsDTO.id,
                    accountId = clientDetailsDTO.accountId,
                    name = clientDetailsDTO.name,
                    finalName = clientDetailsDTO.finalName,
                    age = clientDetailsDTO.age
            )
        }
        return null
    }
}